src/queue/index.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2018  Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import * as Error from '../helpers/errors';
+import amqp from 'amqplib';
+import {isNotDefined} from '../helpers/utils';
+import log from '../helpers/logger';
+
+
+/**
+ * QUEUE_NAME - Queue name for importing data.
+ * @type {string} QUEUE_NAME
+ **/
+export const QUEUE_NAME = 'bookbrainz-import';
+
+/**
+ * Connection - Object containing functions to create and close RMQ connections.
+ * @type {Object} Connection
+ * @property {function} connect - Returns a connection object promise
+ * @property {function} shutdown - Takes in a connection promise and closes the
+ * 		connection
+ **/
+export const Connection = {
+	connect: function connect() {
+		let connection = null;
+		try {
+			connection = amqp.connect('amqp://localhost');
+		}
+		catch (err) {
+			Error.raiseError(Error.CONNECTION_ERROR)(err);
+		}
+		return connection;
+	},
+	shutdown: function shutdown(connectionPromise) {
+		if (isNotDefined(connectionPromise)) {
+			Error.undefinedValue('Connection.shutdown:: connectionPromise');
+			return null;
+		}
+
+		return connectionPromise
```

Comment:
Better to write this with an async function:

    shutdown: async function shutdown(connectionPromise) {
    	if (isNotDefined(connectionPromise)) {
    		Error.undefinedValue('Connection.shutdown:: connectionPromise');
    		return null;
    	}
    
    	try {
    		const connection = await connectionPromise;
    		connection.close();
    	} catch(e) {
    		Error.raiseError(Error.CONNECTION_CLOSE_ERROR)(e);
    	}
    }